### PR TITLE
Issue #1447 - Depend on "fat" GeckoView build from browser-engine-gecko-nightly.

### PR DESCRIPTION
**Pull request by pocmo**
This will unlock some shiny things:
* We can get rid of mirroring the geckoview version in the app projects (Yay!)
* We can stop using a flavor dimension for different architectures (YAY!)
* We can build and deploy AABs.

... only with GeckoView Nightly for now. The other ones will follow later on.

Looking forward to applying some of those optimizations to RB and start experimenting with AABs - before we later on apply that to Fenix.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: automation/taskcluster/androidTest/ui-test.sh [hunk 1]
```diff
@@ -76,14 +76,12 @@ set +e
...
 then
-    deviceType="X86"
     flank_template="$FLANK_CONF_X86"
 else
-    deviceType="Arm"
     flank_template="$FLANK_CONF_ARM"
 fi
 
-APK_APP="./samples/${component}/build/outputs/apk/geckoNightly${deviceType}/debug/samples-browser-geckoNightly-${deviceType,,}-debug.apk"
+APK_APP="./samples/${component}/build/outputs/apk/geckoNightlyUniversal/debug/samples-browser-geckoNightly-universal-debug.apk"
```
Comment (mitchhentges) [inline automation/taskcluster/androidTest/ui-test.sh]: I think you'll need to add `universal` to the list of device types somewhere in the gradle config

Comment (pocmo) [reply automation/taskcluster/androidTest/ui-test.sh]: I am not sure what config that is?

Comment (mitchhentges) [reply automation/taskcluster/androidTest/ui-test.sh]: Err, looks like we already have `universal` as an architecture, never mind :)
https://github.com/mozilla-mobile/android-components/blob/master/samples/browser/build.gradle#L62-L64

File: samples/browser/build.gradle [hunk 2]
```diff
@@ -76,7 +76,13 @@ android {
             setIgnore(true)
         }
 
-        if (flavors.contains("gecko") && flavors.contains("universal")) {
+        // Only allow "universal" flavors of "system" and "geckoviewNightly" engines
+        if (flavors.contains("universal") && !(flavors.contains("system") || flavors.contains("geckonightly"))) {
```
Comment (mitchhentges) [inline samples/browser/build.gradle]: Is this case-sensitive? Does `geckonightly` need to be `geckoNightly`?

Comment (pocmo) [reply samples/browser/build.gradle]: I was wondering about this too, but we join+lowercase above:
`def flavors = variant.flavors*.name.toString().toLowerCase()`

**Pull request merged**